alg/gdalwarper.cpp

```diff
@@ -2486,18 +2486,34 @@ bool GDALGetWarpResampleAlg(const char *pszResampling,
         eResampleAlg = GRA_Q3;
     else if (EQUAL(pszResampling, "sum"))
         eResampleAlg = GRA_Sum;
-    else
+    else if ((pszResampling[0] == 'P' || pszResampling[0] == 'p') &&
+            CPLStrnlen(pszResampling, 16) > 1)
     {
-        if (bThrow)
-        {
-            throw std::invalid_argument("Unknown resampling method");
-        }
-        else
+        const char* pszValue = pszResampling + 1;
+        char* pszEnd = nullptr;
+        const double dfPercentile = CPLStrtod(pszValue, &pszEnd);
+
+        if (pszEnd == pszValue || *pszEnd != '\0' ||
+            dfPercentile <= 0.0 || dfPercentile >= 100.0)
         {
-            CPLError(CE_Failure, CPLE_IllegalArg,
-                     "Unknown resampling method: %s.", pszResampling);
-            return false;
+            if (bThrow)
+            {
+                throw std::invalid_argument(
+                    "Invalid percentile resampling method");
+            }
+            else
+            {
+                CPLError(CE_Failure, CPLE_IllegalArg,
+                        "Invalid percentile resampling method: %s.",
+                        pszResampling);
+                return false;
+            }
         }
+
+        eResampleAlg = GRA_Percentile;
+
+        // Store percentile for warp kernel
+        CPLSetConfigOption("GDAL_WARP_PERCENTILE",
```

Comment:
We shouldn't do that way. Config options are global and messy to deal with. Ideally we would have some C structure describing more about resampling methods than just their identifier., but that would be a quite substantial change. The probably more reasonable way is to request than when GRA_Percentile is set a warping options passed through GDALWarpOperation::papszWarpOptions is specified